src/geometry/interaction_groups.rs

```diff
@@ -23,25 +30,39 @@ pub struct InteractionGroups {
     pub memberships: Group,
     /// Groups filter.
     pub filter: Group,
+    /// Interaction test mode
+    pub test_mode: InteractionTestMode,
+}
+
+#[cfg_attr(feature = "serde-serialize", derive(Serialize, Deserialize))]
+#[derive(Copy, Clone, Debug, Hash, PartialEq, Eq, Default)]
+/// Specifies which method should be used to test interactions
+pub enum InteractionTestMode {
+    /// Use [`InteractionGroups::test_and`].
+    #[default]
+    AND,
+    /// Use [`InteractionGroups::test_or`].
+    OR,
 }
 
 impl InteractionGroups {
     /// Initializes with the given interaction groups and interaction mask.
-    pub const fn new(memberships: Group, filter: Group) -> Self {
+    pub const fn new(memberships: Group, filter: Group, test_mode: InteractionTestMode) -> Self {
         Self {
             memberships,
             filter,
+            test_mode,
         }
     }
```

Comment:
To avoid some API breakage, we could add a `new_with_testmode` constructor, but it's not a very difficult migration so I'm fine with this.

Reply:
I would be inclined to not add it, as it's an ugly API (new_with_testmode), also it hides away this new functionality (people who use new still don't know they can use AND or OR).